### PR TITLE
feat(duration): add formatter options

### PR DESCRIPTION
**Pull request by DesmondWillowbrook**
- add `duration/options.rs`

follow up to #4978

@younies @sffc

### CODE REVIEW

File: components/experimental/src/duration/options.rs [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    year: (YearStyle, UnitVisibility),
+    month: (MonthStyle, UnitVisibility),
+    week: (WeekStyle, UnitVisibility),
+    day: (DayStyle, UnitVisibility),
+    hour: (HourStyle, UnitVisibility),
+    minute: (MinuteStyle, UnitVisibility),
+    second: (SecondStyle, UnitVisibility),
+    millisecond: (MilliSecondStyle, UnitVisibility),
+    microsecond: (MicroSecondStyle, UnitVisibility),
+    nanosecond: (NanoSecondStyle, UnitVisibility),
```
Comment (DesmondWillowbrook) [inline components/experimental/src/duration/options.rs:21]: This seems more ergonomic than having different wrapper structs for each unit, like:
```rs
struct MinuteFormat {
    style: MinuteStyle,
    visibility: UnitVisibility
}
```
Should I instead prefer the above?

Comment (younies) [reply components/experimental/src/duration/options.rs:21]: We can use `struct` for each option such as `Year`, `Month` ... etc.

and you can add a separate file for each struct with its eums.

Comment (sffc) [reply components/experimental/src/duration/options.rs:21]: The options are disjoint in the spec. You can specify `style` and `display`/`visibility` separately.

Comment (DesmondWillowbrook) [reply components/experimental/src/duration/options.rs]: You're right; changed.

**Review by younies — DISMISSED** [review]
Fix clippy and docs

**Review by sffc — APPROVED** [review]

**Review by younies — APPROVED** [review]

**Pull request merged**